subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionPerformanceTest.groovy

```diff
@@ -71,10 +74,30 @@ class AbstractCrossVersionPerformanceTest extends Specification {
         runner
     }
 
+    void applyEnterprisePlugin() {
+        runner.addBuildMutator { invocationSettings ->
+            new ApplyGradleEnterprisePluginMutator(invocationSettings.projectDir)
+        }
+    }
+
     static {
         // TODO - find a better way to cleanup
         System.addShutdownHook {
             RESULTS_STORE.close()
         }
     }
 }
+
+class ApplyGradleEnterprisePluginMutator implements BuildMutator {
```

Comment:
Is there a reason for this type to be public?

Reply:
It is some test infrastructure. I could make it protected, though I think it is also fine to keep it public in here.